### PR TITLE
Add basic SysV init script management of JNLP slave service

### PR DESCRIPTION
**Pull request by foobarbam**
Add support for a very basic SysV style init scripts to better support JNLP slaves in docker containers and *nixes that cannot leverage runit.

Works find in my basic centos docker images as well as the phusion dockers used in the contained test-suites

### CODE REVIEW

File: libraries/plugin.rb [hunk 1]
```diff
@@ -117,14 +117,14 @@ def whyrun_supported?
             new_resource.source,
             new_resource.name,
             nil,
-            cli_opts: new_resource.options,
+            cli_opts: new_resource.options
```
Comment (majormoses) [inline libraries/plugin.rb]: rather than changing all of these I think we just want to update the .rubocop.yml

File: libraries/slave_jnlp.rb [hunk 2]
```diff
@@ -217,6 +220,37 @@ def slave_jar_resource
...
+    #
+    # @return [Chef::Resource::Service]
+    #
+    def sysv_service_resource
+      init_script = ::File.join(::File::SEPARATOR, 'etc',
+                                'init.d', new_resource.service_name)
+
+      sysv_script_template = \
+        Chef::Resource::Template.new(init_script, run_context)
+      sysv_script_template.cookbook('jenkins')
```
Comment (majormoses) [inline libraries/slave_jnlp.rb]: we should param this.

File: libraries/slave_jnlp.rb [hunk 3]
```diff
@@ -217,6 +220,37 @@ def slave_jar_resource
...
+      sysv_script_template.mode('0755')
+      sysv_script_template.source('jenkins-slave-init.erb')
```
Comment (majormoses) [inline libraries/slave_jnlp.rb]: we should param this

**Comment by tas50** [comment]
This looks like something that could potentially be helpful for users, but this PR will need a pretty significant rebase before it's possible to merge anything.

**Comment by Xorima** [comment]
Closing due to inactivity. 

If this is still wanted please reopen or open another pull request. Alternatively drop by the #sous-chefs channel on the [Chef Community Slack](http://community-slack.chef.io/) and we'll be happy to help! 

Thanks,
Sous-Chefs

**Pull request closed without merging**